File: backend/src/domain/financing/schema.rs
#![allow(dead_code)]
use serde::{Deserialize, Serialize};

// Business Plan Record from database
#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct BusinessPlan {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub industry: String,
    pub region: String,
    pub financing_amount: f64,
    pub company_size: String,
    pub created_at: Option<String>,
}

// API Request/Response structures
#[derive(Serialize, Deserialize, Debug)]
pub struct GetBusinessPlanReq {
    pub id: i64,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct GetBusinessPlanRes {
    pub id: i64,
    pub title: String,
    pub content: String,
    pub industry: String,
    pub region: String,
    pub financing_amount: f64,
    pub company_size: String,
    pub created_at: Option<String>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct GetBusinessPlansReq {
    pub ids: Vec<i64>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct CreateBusinessPlanReq {
    pub title: String,
    pub content: String,
    pub industry: String,
    pub region: String,
    pub financing_amount: f64,
    pub company_size: String,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct UpdateBusinessPlanReq {
    pub id: i64,
    pub title: Option<String>,
    pub content: Option<String>,
    pub industry: Option<String>,
    pub region: Option<String>,
    pub financing_amount: Option<f64>,
    pub company_size: Option<String>,
}

#[derive(Serialize, Deserialize, Debug)]
pub struct DeleteBusinessPlanReq {
    pub id: i64,
}

File: backend/domain/financing/schema.go
package financing

import "time"

// BusinessPlan 商业计划实体
type BusinessPlan struct {
	ID              int64      `json:"id"`
	Title           string     `json:"title"`
	Content         string     `json:"content"`
	Industry        string     `json:"industry"`
	Region          string     `json:"region"`
	FinancingAmount float64    `json:"financing_amount"`
	CompanySize     string     `json:"company_size"`
	CreatedAt       *time.Time `json:"created_at"`
	UpdatedAt       *time.Time `json:"updated_at"`
}

// 请求/响应结构体

type GetBusinessPlanReq struct {
	ID int64 `json:"id" binding:"required"`
}

type GetBusinessPlanRes struct {
	ID              int64      `json:"id"`
	Title           string     `json:"title"`
	Content         string     `json:"content"`
	Industry        string     `json:"industry"`
	Region          string     `json:"region"`
	FinancingAmount float64    `json:"financing_amount"`
	CompanySize     string     `json:"company_size"`
	CreatedAt       *time.Time `json:"created_at"`
}

type GetBusinessPlansReq struct {
	IDs []int64 `json:"ids" binding:"required"`
}

type CreateBusinessPlanReq struct {
	Title           string  `json:"title" binding:"required"`
	Content         string  `json:"content" binding:"required"`
	Industry        string  `json:"industry" binding:"required"`
	Region          string  `json:"region" binding:"required"`
	FinancingAmount float64 `json:"financing_amount" binding:"required"`
	CompanySize     string  `json:"company_size" binding:"required"`
}

type UpdateBusinessPlanReq struct {
	ID              int64    `json:"id" binding:"required"`
	Title           *string  `json:"title"`
	Content         *string  `json:"content"`
	Industry        *string  `json:"industry"`
	Region          *string  `json:"region"`
	FinancingAmount *float64 `json:"financing_amount"`
	CompanySize     *string  `json:"company_size"`
}

type DeleteBusinessPlanReq struct {
	ID int64 `json:"id" binding:"required"`
}
